test/e2e/multihoming.go

```diff
@@ -338,27 +381,46 @@ var _ = Describe("Multi Homing", feature.MultiHoming, func() {
 				serverPod := kickstartPod(cs, serverPodConfig)
 
 				By("instantiating the client pod")
-				kickstartPod(cs, clientPodConfig)
+				clientPod := kickstartPod(cs, clientPodConfig)
+
+				serverInterface, err := getNetworkInterfaceName(serverPod, serverPodConfig, netConfig.name)
+				Expect(err).NotTo(HaveOccurred(), "failed to extract server pod interface name")
+
+				clientInterface, err := getNetworkInterfaceName(clientPod, clientPodConfig, netConfig.name)
+				Expect(err).NotTo(HaveOccurred(), "failed to extract client pod interface name")
+
+				// Add external container that will act as external router for the localnet
+				if (clientPodConfig.usesExternalRouter && len(clientPodConfig.attachments) > 0) ||
+					(serverPodConfig.usesExternalRouter && len(serverPodConfig.attachments) > 0) {
+					By("instantiating the external container")
+					externalRouterName, err := createExternalRouter(providerCtx, cs, f, netConfig.vlanID, externalRouterIPOffset)
+					Expect(err).NotTo(HaveOccurred())
+
+					By("injecting routes via the external container")
+					err = injectStaticRoutesViaExternalContainer(f, cs, clientPodConfig, serverPodConfig,
```

Comment:
I think this test can benefit from a step-by-step plain english explanation like here: https://github.com/ovn-kubernetes/ovn-kubernetes/blob/56d14a3074ba7df00a6b68dfe4ac60aff7fc624a/test/e2e/egressip.go#L865 but including details around the nbctl routes you add and why.
having the step by step comments can help the next contributor who needs to modify this test and understand why we did the setup this way etc